Release note 1.2 (#4469)
istio/istio.io: content/boilerplates/notes/1.2.md

- **Improved** Sidecar support for NONE mode (without iptables).
- **Added** support for configuring the paths for Istio mutual TLS certificates.
- **Added** ability to configure the DNS refresh rate for sidecar Envoys, to reduce the load on the DNS servers.
- **Added** `traffic.sidecar.istio.io/includeInboundPorts` annotation to eliminate the need for service owner to declare `containerPort` in the deployment yaml file.

## Security

- **Added** Support for [PKCS 8](https://en.wikipedia.org/wiki/PKCS_8) private keys for workloads, enabled by the field `pkcs8-keys` on Citadel.
- **Fixed** SAN field in workload certificates is set as `critical`.
- **Improved** Added Jitter to Citadel Agent for sending CSR requests.
- **Improved** Extended the lifetime of self-signed Citadel root certificates to 10 years.
- **Improved** JWT public key fetching logic to make it more resilient to network failure.

## Telemetry

- **Added** Envoy metrics support to include metrics by prefixes, suffixes and regular expressions.
- **Changed** Prometheus generated traffic is excluded from metrics.

## Policy

- **Fixed** TCP Policy enforcement.

## Configuration Management

> - **Changed** Galley File System reader code to be require explicit reload signal, instead of automatic detection. The reload can be triggered via SIGUSR1 or through ControlZ.

Reviewer: I don't think we want this item in user-facing doc, let's remove.

> - **Made** legacy Mixer CRDs optional for Galley startup precondition checks.

Reviewer: Not interesting, remove

> - **Fixed** a race condition that could cause a deadlock during graceful shutdown of Galley.

Reviewer: Remove, not interesting

- **Added** ControlZ support to visualize the state of the MCP Server in Galley.
- **Enabled** validation of more Mixer CRDs in Galley validation webhook.

> - **Added** support for specifying connection metadata for MCP dial-out mode in Galley.

Reviewer: @ozevren Is there a link to provide here in user docs?

Author: Sorry hit resolved on the wrong comment.  this is still needed:  @ozevren Is there a link to provide here in user docs?

> ## Installation

Reviewer 2: Ordering of categories is odd. Would have

General
Networking
Policy & Security 
Telemetry
Install & Tools
APIs & Configuration Management

- **Added** pod anti-affinity and toleration support to all of our control plane components.
- **Added** `sidecarInjectorWebhook.neverInjectSelector` and `sidecarInjectorWebhook.alwaysInjectSelector` to allow users to further refine workloads that have sidecar automatically injected or not, based on label selectors.
- **Added** `global.logging.level` and `global.proxy.logLevel` to allow users to easily configure logs for control plane and data plane components globally.
- **Added** support to configure the Datadog location via `global.tracer.datadog.address`.

## Upgrades

> - **Removed** deprecated Adapter and Template CRDs by default. Configure `mixer.templates.useTemplateCRDs=true` and `mixer.adapters.useAdapterCRDs=true` to re-enable them.

Reviewer 2: @mandarjog 

This needs more exposition

Author: This is also described more in the upgrade notice:  https://github.com/istio/istio.io/pull/4471

Reviewer 3: we could also link to the deprecation notice, if so desired: https://discuss.istio.io/t/deprecation-notice-custom-mixer-adapter-crds/2055

Author: added

> - **Removed** deprecated Adapter and Template CRDs by default. Configure `mixer.templates.useTemplateCRDs=true` and `mixer.adapters.useAdapterCRDs=true` to re-enable them.

Reviewer 4: ```suggestion
- **Removed** Previously [deprecated]( https://discuss.istio.io/t/deprecation-notice-custom-mixer-adapter-crds/2055) Adapter and Template CRDs are disabled by default. You may use  `mixer.templates.useTemplateCRDs=true` and `mixer.adapters.useAdapterCRDs=true` install options to re-enable them.
```

Author: accepted

Refer to the [installation option change page](/docs/reference/config/installation-options-changes/) to view the complete list of changes.

## General

> - **Added** `traffic.sidecar.istio.io/includeInboundPorts` annotation to eliminate the need for service owner to declare `containerPort` in the deployment yaml file.

Reviewer 2: Need to document whether this will become the default in the next release. Want to encourage users to do this.

Author: Done

- **Introduced** IPv6 alpha support for Kubernetes clusters.

## `istioctl` and `kubectl`

> - **Graduated** `istioctl verify-install` out of experimental.

Reviewer 2: We should try to order sections in terms of user impact

Author: done

- **Improved** `istioctl verify-install` to validate if a given Kubernetes environment meets Istio's prerequisites.
- **Added** auto-completion support to `istioctl`.
- **Added** `istioctl experimental dashboard` to allow users to easily open the web UI of any Istio addons.
- **Added** `istioctl x` alias to conveniently run `istioctl experimental` command.
- **Improved** `istioctl version` to report both Istio control plane and `istioctl` version info by default.
- **Improved** `istioctl validate` to validate Mixer configuration and supports deep validation with referential integrity.

## Traffic Management

- **Improved** locality based routing in multicluster environments.

> - **Improved** outbound traffic policy in `ALLOW_ANY` mode. Traffic for unknown HTTP/HTTPS hosts on an existing port will be forwarded as is. Unknown traffic will be logged in Envoy access logs.

Reviewer 2: Link to doc for ALLOW_ANY and indicate what the default behavior is

Reviewer 5: https://istio.io/docs/tasks/traffic-management/egress/#envoy-passthrough-to-external-services

Author: added link

> - **Added** support for sending traces to Datadog.

Reviewer 2: This is called out above. Remove this and put into Telemetry

Author: moved.

- **Added** support for setting HTTP idle timeouts to upstream services.

> - **Improved** Sidecar support for NONE mode (without iptables).

Reviewer 2: Needs more exposition

Author: added a link

> - **Added** support for configuring the paths for Istio mutual TLS certificates.

Reviewer 2: link to docs

Author: pinged @myidpt

- **Added** ability to configure the DNS refresh rate for sidecar Envoys, to reduce the load on the DNS servers.

## Security

> - **Added** Kubernetes health check prober rewrite per deployment via `sidecar.istio.io/rewriteAppHTTPProbers: "true"` in the `PodSpec` annotation.

Reviewer 2: Needs link

Author: added

- **Fixed** mutual TLS probe rewrite for HTTPS probes

> - **Added** Support for [PKCS 8](https://en.wikipedia.org/wiki/PKCS_8) private keys for workloads, enabled by the field `pkcs8-keys` on Citadel.

Reviewer 2: @wattli @wenchenglu 
I don't know what a 'field' on Citadel is

Reviewer 6: ```suggestion
- **Added** Support for [PKCS 8](https://en.wikipedia.org/wiki/PKCS_8) private keys for workloads, enabled by the `pkcs8-keys` flag on Citadel.
```

Author: updated

> - **Fixed** SAN field in workload certificates is set as `critical`.

Reviewer 2: Needs a link. What does 'critical' mean?

Reviewer 2: @wenchenglu

Reviewer 6: Probably need to link to [RFC5280, section 4.2](https://tools.ietf.org/html/rfc5280#section-4.2)

> Each extension in a certificate is designated as either critical or non-critical.  A certificate-using system MUST reject the certificate if it encounters a critical extension it does not recognize or a critical extension that contains information that it cannot process.  A non-critical extension MAY be ignored if it is not recognized, but MUST be processed if it is recognized.

Author: someone updated!!

> - **Improved** Added Jitter to Citadel Agent for sending CSR requests.

Reviewer 2: reword...
added jitter when sending CSR requests for certificate generation

Author: removed this as this is delivered in 1.1.x in early april.

> - **Improved** Extended the lifetime of self-signed Citadel root certificates to 10 years.

Reviewer 2: s/Extended/extend

> - **Improved** JWT public key fetching logic to make it more resilient to network failure.

Reviewer 2: reword

make JWT public key fetching logic more resilient to network failure

Author: reworded

## Telemetry

> - **Added** Envoy metrics support to include metrics by prefixes, suffixes and regular expressions.

Reviewer 2: @douglas-reid 

Do we mean

'added support for filtering Envoy metrics by prefix, suffix or regex'

?

Reviewer 3: It is both filtering and including more (from our heavily-filtered set).

`**Added** Full support for control over Envoy stats generation, based on stats prefixes, suffixes, and regular expressions` ?

Reviewer 4: @douglas-reid 's suggestion sounds good.

Author: accepted

> - **Added** Envoy metrics support to include metrics by prefixes, suffixes and regular expressions.

Reviewer 4: ```suggestion
- **Added** Full support for control over Envoy stats generation, based on stats prefixes, suffixes, and regular expressions through the use of [annotations](https://preliminary.istio.io/docs/reference/commands/pilot-agent/#annotations).
```

Author: accepted

- **Changed** Prometheus generated traffic is excluded from metrics.

## Policy

> - **Fixed** TCP Policy enforcement.

Reviewer 2: @liminw 

Is this a fix or add ? If it's a fix can we link to the tracking issue?

Reviewer 4: Fixed https://github.com/istio/istio/issues/13868

Reviewer 2: @liminw Does this affect RBAC?

Reviewer 4: It does not affect RBAC, it only affects TCP policy enforcement thru Mixer.

Reviewer 4: **Fixed** [Mixer based ](https://github.com/istio/istio/issues/13868)TCP Policy enforcement.

Author: updated

## Configuration Management

> - **Added** ControlZ support to visualize the state of the MCP Server in Galley.

Reviewer 2: Move down

Author: updated

> - **Enabled** validation of more Mixer CRDs in Galley validation webhook.

Reviewer 2: Move up, this is primary user facing

Author: updated

> - **Added** an explicit command-line flag to control the service discovery module in Galley.

Reviewer 2: @ozevren 

Link ?

Author: updated

> - **Introduced** environment variable and configuration file support for configuring Galley, in addition to command-line flags.

Reviewer 2: @ozevren 

Can we link to any doc. Should move up in list

Reviewer 7: https://docs.google.com/document/d/1M-qqBMNbhbAxl3S_8qQfaeOLAiRqSBpSgfWebFBRuu8/edit

Reviewer 7: Not sure how editing of this supposed to work. @linsun Will you be making these changes?

Author: yes done.  Thanks!

> - **Introduced** environment variable and configuration file support for configuring Galley, in addition to command-line flags.

Reviewer 2: @linsun 

What is the difference between "Added" & "Introduced"

Author: I'm changing all to "Added"

> - **Added** `InitialWindowSize` and `InitialConnWindowSize` parameters to Galley and Pilot to allow fine-tuning of MCP (gRPC) connection settings.

Reviewer 2: I think we should move things like this into an "Other" category as they are only barely user-facing

Author: Done

- **Added** ability to configure the DNS refresh rate for sidecar Envoys, to reduce the load on the DNS servers.

## Security

- **Added** Kubernetes health check prober rewrite per deployment via `sidecar.istio.io/rewriteAppHTTPProbers: "true"` in the `PodSpec` annotation.
- **Added** Support for [PKCS 8](https://en.wikipedia.org/wiki/PKCS_8) private keys for workloads, enabled by the field `pkcs8-keys` on Citadel.
- **Fixed** SAN field in workload certificates is set as `critical`.
- **Improved** Added Jitter to Citadel Agent for sending CSR requests.
- **Improved** Extended the lifetime of self-signed Citadel root certificates to 10 years.
- **Improved** JWT public key fetching logic to make it more resilient to network failure.

## Telemetry

- **Added** Envoy metrics support to include metrics by prefixes, suffixes and regular expressions.
- **Changed** Prometheus generated traffic is excluded from metrics.

## Policy

- **Fixed** TCP Policy enforcement.

## Configuration Management

- **Added** ControlZ support to visualize the state of the MCP Server in Galley.
- **Enabled** validation of more Mixer CRDs in Galley validation webhook.

> - **Added** support for specifying connection metadata for MCP dial-out mode in Galley.

Author: Sorry hit resolved on the wrong comment. this is still needed: @ozevren Is there a link to provide here in user docs?

Reviewer 7: We have command-line flags documented, but not a more comprehensive doc. In either case, this is a very rough-cut alpha-like feature at this point.

Reviewer 7: https://preliminary.istio.io/docs/reference/commands/galley/#galley-server

Author: discussed with @ozevren offline, think this would not be used by users directly, and we should remove it.

## Installation

> - **Added** `global.proxy.resources.limits.memory` to allow users to limit the memory of sidecar proxies.

Reviewer 5: We just changed the default I think

Reviewer 8: Then let's remove this from the release notes unless @rshriram objects

Reviewer 9: Might be worth calling out the changed defaults AND that it's configurable through installation

Author: oh, i guess the diff tool was not accurate, I will remove it.

Author: added back

- **Added** pod anti-affinity and toleration support to all of our control plane components.
- **Added** `sidecarInjectorWebhook.neverInjectSelector` and `sidecarInjectorWebhook.alwaysInjectSelector` to allow users to further refine workloads that have sidecar automatically injected or not, based on label selectors.
- **Added** `global.logging.level` and `global.proxy.logLevel` to allow users to easily configure logs for control plane and data plane components globally.
- **Added** support to configure the Datadog location via `global.tracer.datadog.address`.

## Upgrades

- **Removed** deprecated Adapter and Template CRDs by default. Configure `mixer.templates.useTemplateCRDs=true` and `mixer.adapters.useAdapterCRDs=true` to re-enable them.

Refer to the [installation option change page](/docs/reference/config/installation-options-changes/) to view the complete list of changes.

## General

- **Added** `traffic.sidecar.istio.io/includeInboundPorts` annotation to eliminate the need for service owner to declare `containerPort` in the deployment yaml file.

> - **Introduced** IPV6 alpha support for Kubernetes clusters.

Reviewer 8: Nit:  IPv6

## `istioctl` and `kubectl`

- **Graduated** `istioctl verify-install` out of experimental.
- **Improved** `istioctl verify-install` to validate if a given Kubernetes environment meets Istio's prerequisites.
- **Added** auto-completion support to `istioctl`.
- **Added** `istioctl experimental dashboard` to allow users to easily open the web UI of any Istio addons.
- **Added** `istioctl x` alias to conveniently run `istioctl experimental` command.
- **Improved** `istioctl version` to report both Istio control plane and `istioctl` version info by default.
- **Improved** `istioctl validate` to validate Mixer configuration and supports deep validation with referential integrity.

## Traffic Management

> - **Improved** locality based routing in multicluster environments.

Reviewer 8: @howardjohn @liamawhite I think we can say something stronger here.  We made locality load balancing actually work.

We also added it as a Beta feature in PR https://github.com/istio/istio.io/pull/4434

Reviewer 2: Do we have docs? If so can we link

Reviewer 5: https://istio.io/help/ops/traffic-management/locality-load-balancing/

Author: Thanks, found the link for 1.2 also.

## General

- **Added** `traffic.sidecar.istio.io/includeInboundPorts` annotation to eliminate the need for service owner to declare `containerPort` in the deployment yaml file.  This will become the default in a future release.

> - **Added** IPv6 alpha support for Kubernetes clusters.

Reviewer 8: Or even experimental.  I'm worried that this one does not have any test coverage

Author: doe

- **Improved** outbound traffic policy in `ALLOW_ANY` mode (lin to doc). Traffic for unknown HTTP/HTTPS hosts on an existing port will be forwarded as is. Unknown traffic will be logged in Envoy access logs.
- **Added** support for setting HTTP idle timeouts to upstream services.
- **Improved** Sidecar support for NONE mode (without iptables) (link to doc).
- **Added** ability to configure the DNS refresh rate (need a link) for sidecar Envoys, to reduce the load on the DNS servers.

## Security

- **Improved** Extended the lifetime of self-signed Citadel root certificates to 10 years.
- **Added** Kubernetes health check prober rewrite per deployment via `sidecar.istio.io/rewriteAppHTTPProbers: "true"` in the `PodSpec` annotation. (link to doc)
- **Added** support for configuring the secrete paths for Istio mutual TLS certificates.
- **Added** support for [PKCS 8](https://en.wikipedia.org/wiki/PKCS_8) private keys for workloads, enabled by the field `pkcs8-keys` on Citadel.
- **Added** jitter(need a link) when sending CSR requests for certification generation.
- **Improved** JWT public key fetching logic to make it more resilient to network failure.

> - **Fixed** SAN field in workload certificates is set as `critical`. - need clear explanation

Reviewer 10: ```suggestion
- **Fixed** [SAN](https://tools.ietf.org/html/rfc5280#section-4.2.1.6) field in workload certificates is set as `critical`. This is required because the `subject` field is empty for Istio workload certificates.
```

- **Fixed** mutual TLS probe rewrite for HTTPS probes.

## Telemetry

> - **Added** Envoy metrics support to include metrics by prefixes, suffixes and regular expressions. (rewrite and link?)

Reviewer 4: ```suggestion
- - **Added** Full support for control over Envoy stats generation, based on stats prefixes, suffixes, and regular expressions through the use of [annotations](https://preliminary.istio.io/docs/reference/commands/pilot-agent/#annotations).
```

> - **Fixed** [SAN](https://tools.ietf.org/html/rfc5280#section-4.2.1.6) field in workload certificates is set as `critical`. This is required because the Istio workload certificates have empty `subject` fields.

Reviewer 10: ```suggestion
- **Fixed** [SAN](https://tools.ietf.org/html/rfc5280#section-4.2.1.6) field in workload certificates is set as `critical`. This indicates that the certificate verifiers MUST verify the SAN field in the certificates. This is required because the Istio workload certificates have empty `subject` fields.
```

- **Changed** Prometheus generated traffic is excluded from metrics.
- **Added** support for sending traces to Datadog.

## Policy

> - **Fixed** TCP Policy enforcement in Mixer.

Reviewer 4: ```suggestion
- **Fixed** [Mixer based ](https://github.com/istio/istio/issues/13868)TCP Policy enforcement.
```

- Any channge in istio RBAC?

## Configuration Management

> - **Enabled** webhook validation of more CRDs.

Reviewer 7: ```suggestion
- **Improved** validation of Policy & Telemetry CRDs.
```

- **Added** support to configure the Datadog location via `global.tracer.datadog.address`. (link to the page)
- **Removed** [deprecated Adapter and Template CRDs](https://discuss.istio.io/t/deprecation-notice-custom-mixer-adapter-crds/2055). Configure `mixer.templates.useTemplateCRDs=true` and `mixer.adapters.useAdapterCRDs=true` to re-enable them.

Refer to the [installation option change page](/docs/reference/config/installation-options-changes/) to view the complete list of changes.

## `istioctl` and `kubectl`

- **Graduated** `istioctl verify-install` out of experimental.
- **Improved** `istioctl verify-install` to validate if a given Kubernetes environment meets Istio's prerequisites.
- **Added** auto-completion support to `istioctl`.
- **Added** `istioctl experimental dashboard` to allow users to easily open the web UI of any Istio addons.
- **Added** `istioctl x` alias to conveniently run `istioctl experimental` command.
- **Improved** `istioctl version` to report both Istio control plane and `istioctl` version info by default.
- **Improved** `istioctl validate` to validate Mixer configuration and supports deep validation with referential integrity.

## Other

> 

Reviewer 11: Add an item about Istio CNI:
```
- **Added** Istio CNI support to setup sidecar network redirection and remove the use of `istio-init` containers requiring `NET_ADMIN` capability.
```

I don't see any notes about limitations / known issues in this doc, but I'd document one known issue about Istio CNI in 1.2:
```
Known issue: Istio CNI doesn't honor `global.proxy.*` redirection settings.
```

Reviewer 11: @louiscryan prefers to leave the known issue out of this doc.

Author: @rlenglet can I use improved instead? (since this is in 1.1)  It would be good to highlight what has been improved.

Reviewer 11: @linsun It seems like it was not included in any release notes for 1.1.x releases, which is why I thought it'd be appropriate to word it this way.

Author: WOW, I know it is been in 1.1 docs... perhaps not highlighted in relnnote

Author: added

- **Improved** outbound traffic policy in [`ALLOW_ANY` mode](/docs/reference/config/installation-options/#global-options). Traffic for unknown HTTP/HTTPS hosts on an existing port will be [forwarded as is](/docs/tasks/traffic-management/egress/egress-control/#envoy-passthrough-to-external-services). Unknown traffic will be logged in Envoy access logs.
- **Added** support for setting HTTP idle timeouts to upstream services.
- **Improved** Sidecar support for [NONE mode](/docs/reference/config/networking/v1alpha3/sidecar/#CaptureMode) (without iptables) .
- **Added** ability to configure the [DNS refresh rate](/docs/reference/config/installation-options/#global-options) for sidecar Envoys, to reduce the load on the DNS servers.
- **Graduated** [Sidecar API](/docs/reference/config/networking/v1alpha3/sidecar/) from Alpha to Alpha API and Beta runtime.

## Security

- **Improved** the lifetime of self-signed Citadel root certificates to 10 years.
- **Added** Kubernetes health check prober rewrite per deployment via `sidecar.istio.io/rewriteAppHTTPProbers: "true"` in the `PodSpec` [annotation](/docs/ops/setup/app-health-check/#use-annotations-on-pod).

> - **Added** support for configuring the secrete paths for Istio mutual TLS certificates.

Reviewer 12: s/secrete/secret ?

Author: thx

- **Improved** outbound traffic policy in [`ALLOW_ANY` mode](/docs/reference/config/installation-options/#global-options). Traffic for unknown HTTP/HTTPS hosts on an existing port will be [forwarded as is](/docs/tasks/traffic-management/egress/egress-control/#envoy-passthrough-to-external-services). Unknown traffic will be logged in Envoy access logs.
- **Added** support for setting HTTP idle timeouts to upstream services.
- **Improved** Sidecar support for [NONE mode](/docs/reference/config/networking/v1alpha3/sidecar/#CaptureMode) (without iptables) .
- **Added** ability to configure the [DNS refresh rate](/docs/reference/config/installation-options/#global-options) for sidecar Envoys, to reduce the load on the DNS servers.
- **Graduated** [Sidecar API](/docs/reference/config/networking/v1alpha3/sidecar/) from Alpha to Alpha API and Beta runtime.

## Security

- **Improved** the lifetime of self-signed Citadel root certificates to 10 years.
- **Added** Kubernetes health check prober rewrite per deployment via `sidecar.istio.io/rewriteAppHTTPProbers: "true"` in the `PodSpec` [annotation](/docs/ops/setup/app-health-check/#use-annotations-on-pod).

> - **Added** support for configuring the secrete paths for Istio mutual TLS certificates.

Reviewer 10: @wattli do you know any links for this?

Reviewer 13: no, i don't think we have, the flag is .Values.global.sds.customTokenDirectory

Author: thanks, added the flag

- **Added** support for [PKCS 8](https://en.wikipedia.org/wiki/PKCS_8) private keys for workloads, enabled by the flag `pkcs8-keys` on Citadel.
- **Improved** JWT public key fetching logic to be more resilient to network failure.

> - **Fixed** [SAN](https://tools.ietf.org/html/rfc5280#section-4.2.1.6) field in workload certificates is set as `critical`. This indicates that the certificate verifiers MUST verify the SAN field in the certificates. This is required because the Istio workload certificates have empty `subject` fields.

Reviewer 10: ```suggestion
- **Fixed** [SAN](https://tools.ietf.org/html/rfc5280#section-4.2.1.6) field in workload certificates is set as `critical`. This fixes the issue that some custom certificate verifiers cannot verify Istio certificates.
```

- **Improved** outbound traffic policy in [`ALLOW_ANY` mode](/docs/reference/config/installation-options/#global-options). Traffic for unknown HTTP/HTTPS hosts on an existing port will be [forwarded as is](/docs/tasks/traffic-management/egress/egress-control/#envoy-passthrough-to-external-services). Unknown traffic will be logged in Envoy access logs.
- **Added** support for setting HTTP idle timeouts to upstream services.
- **Improved** Sidecar support for [NONE mode](/docs/reference/config/networking/v1alpha3/sidecar/#CaptureMode) (without iptables) .
- **Added** ability to configure the [DNS refresh rate](/docs/reference/config/installation-options/#global-options) for sidecar Envoys, to reduce the load on the DNS servers.
- **Graduated** [Sidecar API](/docs/reference/config/networking/v1alpha3/sidecar/) from Alpha to Alpha API and Beta runtime.

## Security

- **Improved** extend the default lifetime of self-signed Citadel root certificates to 10 years.
- **Added** Kubernetes health check prober rewrite per deployment via `sidecar.istio.io/rewriteAppHTTPProbers: "true"` in the `PodSpec` [annotation](/docs/ops/setup/app-health-check/#use-annotations-on-pod).

> - **Added** support for configuring the secret paths via `global.sds.customTokenDirectory` for Istio mutual TLS certificates. Refer [here](https://github.com/istio/istio/issues/11984) for more details.

Reviewer 10: ```suggestion
- **Added** support for configuring the secret paths for Istio mutual TLS certificates. Refer [here](https://github.com/istio/istio/issues/11984) for more details.
```

Reviewer 10: `global.sds.customTokenDirectory` is to configure the JWT token loading path. Not related to this feature.

- **Added** IPv6 experimental support for Kubernetes clusters.

## Traffic Management

- **Improved** [locality based routing](/docs/ops/traffic-management/locality-load-balancing/) in multicluster environments.
- **Improved** outbound traffic policy in [`ALLOW_ANY` mode](/docs/reference/config/installation-options/#global-options). Traffic for unknown HTTP/HTTPS hosts on an existing port will be [forwarded as is](/docs/tasks/traffic-management/egress/egress-control/#envoy-passthrough-to-external-services). Unknown traffic will be logged in Envoy access logs.
- **Added** support for setting HTTP idle timeouts to upstream services.
- **Improved** Sidecar support for [NONE mode](/docs/reference/config/networking/v1alpha3/sidecar/#CaptureMode) (without iptables) .
- **Added** ability to configure the [DNS refresh rate](/docs/reference/config/installation-options/#global-options) for sidecar Envoys, to reduce the load on the DNS servers.
- **Graduated** [Sidecar API](/docs/reference/config/networking/v1alpha3/sidecar/) from Alpha to Alpha API and Beta runtime.

## Security

> - **Improved** the lifetime of self-signed Citadel root certificates to 10 years.

Reviewer 10: ```suggestion
- **Improved** extend the default lifetime of self-signed Citadel root certificates to 10 years.
```

- **Added** Kubernetes health check prober rewrite per deployment via `sidecar.istio.io/rewriteAppHTTPProbers: "true"` in the `PodSpec` [annotation](/docs/ops/setup/app-health-check/#use-annotations-on-pod).

> - **Added** support for configuring the secret paths for Istio mutual TLS certificates.

Reviewer 10: ```suggestion
- **Added** support for configuring the secret paths for Istio mutual TLS certificates. Details described in Github [issue](https://github.com/istio/istio/issues/11984).
```
